lib/web_console/whiny_object.rb

```diff
@@ -0,0 +1,23 @@
+module WebConsole
+  class WhinyObject < SimpleDelegator
```

Comment:
I finally put hands on the third option, so I added a new commit implementing that: https://github.com/rails/web-console/commit/60eb494473a110af7bb6b235a7a84efb20d77b3a

I really like the result, no more specific classes doing only the log decision part. The responsibility went to Response and Request classes, with a little part into the Middleware that is taking care of initializing the logger object and passing along to the other collaborators.